File: car-listing-service/rental/core/entities/cars.py
from entities.base_entity import BaseEntity


class Cars(BaseEntity):
    def __init__(self,
                 car_id: int,
                 car_code: str,
                 name: str,
                 year: str,
                 passenger: int,
                 transmission: str,
                 luggage_large: int,
                 luggage_small: int,
                 engine: str,
                 fuel: str,
                 status: str,
                 created_at=None, created_by=None, updated_at=None, updated_by=None):

        super().__init__(created_at, created_by, updated_at, updated_by)

        self.car_id = car_id
        self.car_code = car_code
        self.name = name
        self.year = year
        self.passenger = passenger
        self.transmission = transmission
        self.luggage_large = luggage_large
        self.luggage_small = luggage_small
        self.engine = engine
        self.fuel = fuel
        self.status = status

File: car-listing-service/src/main/java/io/classiccrew/car_listing_service/entity/Cars.java
package io.classiccrew.car_listing_service.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Cars extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    private Long carId;
    private String carCode;
    private String name;
    private String modelYear;
    private int passenger;
    private String transmission;
    private int luggageLarge;
    private int luggageSmall;
    private String engine;
    private String fuel;
    private String status;
}
